docs/copied-from-beats/docs/security/users.asciidoc

```diff
@@ -197,156 +344,245 @@ information.
-----
-endif::no_ilm[]
+{kib} users typically need to view apps, dashboards, and visualizations that contain
+APM data. These users might also need to create and edit dashboards, visualizations, and machine learning jobs.
 
-To grant the required privileges:
+To grant users the required privileges:
 
-. Create a *writer role*, called something like +{beat_default_index_prefix}_writer+,
-that has the following privileges:
+. Assign the following built-in roles to users who need to read APM data:
++
+[options="header"]
+|====
+|Role | Purpose
+
+|`kibana_user`
+|Use the APM UI
+
+|`apm_user`
+|Use the APM UI
+|====
 +
-NOTE: The `monitor` cluster privilege and the `create_doc` privilege on
-+{beat_default_index_prefix}-*+ indices are required in every configuration.
+. If you're using any of the APM, Dashboard, Visualize, Discover, etc., apps,
+you'll need assign space privileges:
 +
 [options="header"]
 |====
 |Type | Privilege | Purpose
 
-ifndef::apm-server[]
-|Cluster
-|`monitor`
-|Retrieve cluster details (e.g. version) 
-endif::apm-server[]
+| Spaces
+| `Read` or `All` on {beat_kib_app}
+| Allow the use of the {beat_kib_app}
 
-ifndef::no_ilm[]
-|Cluster
-|`read_ilm`
-| Read the ILM policy when connecting to clusters that support ILM.
-Not needed when `setup.ilm.check_exists` is `false`.
-endif::no_ilm[]
+| Spaces
+| `Read` or `All` on Dashboards, Visualize, and Discover
+| Allow the user to view, edit, and create dashboards, as well as browse data.
+|====
++
+. Finally, assign the following role if a user needs to enable and edit machine learning features:
++
+[options="header"]
+|====
+|Role | Purpose
 
-ifeval::["{beatname_lc}"=="filebeat"]
-|Cluster
-|`cluster:admin/ingest/pipeline/get`
-|Check for ingest pipelines used by modules. Needed when using modules.
-endif::[]
+|`machine_learning_admin`
+|NEED DESCRIPTION
+|====
 
-|Index
-|`create_doc` on +{beat_default_index_prefix}-*+ indices
-|Write events into {es}
+////
+***********************************  ***********************************
+***********************************  ***********************************
+////
 
-ifndef::no_ilm[]
-|Index
-|`view_index_metadata` on +{beat_default_index_prefix}-*+ indices
-|Check for alias when connecting to clusters that support ILM.
-Not needed when `setup.ilm.check_exists` is `false`.
-endif::no_ilm[]
+[[privileges-agent-central-config]]
+=== Grant privileges and roles needed for APM Agent central configuration
 
-|Index
-|`create_index` on +{beat_default_index_prefix}-*+ indices
-|Create daily indices when connecting to clusters that do not support ILM.
-Not needed when using ILM.
-|====
-ifndef::apm-server[]
-+
-Omit any privileges that aren't relevant in your environment.
-endif::apm-server[]
+++++
+<titleabbrev>Create a _central config_ user</titleabbrev>
+++++
 
-. Assign the *writer role* to users who will index events into {es}.
+[[privileges-agent-central-config-server]]
+==== APM Server central configuration management
```

Comment:
Users might not be super familiar with the internals, that an APM agent retrieves the updated agent configuration via APM Server when they change it in Kibana. Therefore some explanation about this would be great, something like: When configuring central configurations for your agents in Kibana, ensure the user that you configure in APM Server to talk to Kibana has this role.